sdk/src/main/java/com/kin/ecosystem/Kin.java

```diff
@@ -199,25 +200,48 @@ private static void checkInstanceNotNull() throws ClientException {
 	 * Launch Kin Marketplace if the user is activated, otherwise it will launch Welcome to Kin page.
 	 *
 	 * @param activity the activity user can go back to.
+	 * @deprecated Please use {@link Kin#launchEcosystem}
 	 */
+	@Deprecated
 	public static void launchMarketplace(@NonNull final Activity activity) throws ClientException {
+		checkInstanceNotNull();
+		instance.eventLogger.send(EntrypointButtonTapped.create());
+		launchEcosystem(activity, EcosystemExperience.MARKETPLACE);
+	}
+
+	/**
+	 * Launch a specific experience in KinEcosystem if the user is activated,
+	 * otherwise it will launch Welcome to Kin page and then the experience.
+	 *
+	 * @param activity the activity user can go back to.
+	 * @param experience should be one of {@link EcosystemExperience}
+	 */
+	public static void launchEcosystem(@NonNull final Activity activity, @EcosystemExperience final int experience)
+		throws ClientException {
 		checkInstanceNotNull();
 		instance.eventLogger.send(EntrypointButtonTapped.create());
```

Comment:
need to update BI - could be in a different task

Reply:
well, right now its the entry point, rather its order history or marketplace, but maybe we should update it later.